src/query/procedure/module.cpp

```diff
@@ -995,11 +1016,41 @@ bool PythonModule::Close() {
   procedures_.clear();
   transformations_.clear();
   functions_.clear();
-  // Delete the module from the `sys.modules` directory so that the module will
-  // be properly imported if imported again.
+
+  // Get the reference to sys.modules dictionary
   py::Object sys(PyImport_ImportModule("sys"));
-  if (PyDict_DelItemString(sys.GetAttr("modules").Ptr(), file_path_.stem().c_str()) != 0) {
-    spdlog::warn("Failed to remove the module from sys.modules");
+  PyObject *sys_mod_ref = sys.GetAttr("modules").Ptr();
+
+  std::filesystem::path submodules_path = file_path_.parent_path();
+  std::string_view stem = std::string_view(file_path_.stem().c_str());
+  submodules_path /= "mage";
```

Comment:
Why is `mage` hardcoded here?

Reply:
If it's unavoidable, add it as a runtime flag user can change with `mage` as the default value 🤔

Reply:
Done